game/console.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require_relative 'loader'
+
+class Console
+  include ConsoleGame
+  WIN_MATRIX = '++++'
+
+  def run
+    loop do
+      View.menu
+      case View.fetch_user_input
+      when I18n.t('menu.start') then start
+      when I18n.t('menu.rules') then View.rules
+      when I18n.t('menu.statistics') then View.statistics
+      when I18n.t('menu.exit') then exit
+      else View.menu_message_error end
+    end
+  end
+
+  def start
+    @codebraker_game = ::Codebraker::Game.new
+    registration
+    @codebraker_game.start
+    game
+  end
+
+  def registration
+    name_input
+    input_difficulty
+  end
+
+  def name_input
+    loop do
+      name = View.obtain_name
+      begin
+        @codebraker_game.user.name = name
+        break
+      rescue LengthError
+        View.name_length_error
+      end
+    end
+  end
+
+  def input_difficulty
+    loop do
+      difficulty = View.obtain_difficulty
+      if difficulty_accessible?(difficulty)
+        @codebraker_game.difficulty = difficulty.to_sym
+        break
+      else
+        View.difficulty_input_error
+      end
+    end
+  end
+
+  def game
+    loop { break if lost? || procced? }
+    new_game
+  end
+
+  def procced?
+    case guess = View.obtain_guess
+    when I18n.t('menu.exit') then exit
+    when I18n.t('game.hint')
+      hint
+      false
+    else guess_passed?(guess) end
```

Comment:
Method with question mark has to return only true/false result. Why it can exit from game?